http/handler_tx.go

```diff
@@ -57,41 +55,55 @@ func (h *txHandler) NewConcurrentTxn(rw http.ResponseWriter, req *http.Request)
 }
 
 func (h *txHandler) Commit(rw http.ResponseWriter, req *http.Request) {
-	txs := req.Context().Value(txsContextKey).(*sync.Map)
+	txs := mustGetContextSyncMap(req)
 
-	txId, err := strconv.ParseUint(chi.URLParam(req, "id"), 10, 64)
+	txID, err := strconv.ParseUint(chi.URLParam(req, "id"), 10, 64)
 	if err != nil {
 		responseJSON(rw, http.StatusBadRequest, errorResponse{ErrInvalidTransactionId})
 		return
 	}
-	txVal, ok := txs.Load(txId)
+	txVal, ok := txs.Load(txID)
 	if !ok {
 		responseJSON(rw, http.StatusBadRequest, errorResponse{ErrInvalidTransactionId})
 		return
 	}
-	err = txVal.(datastore.Txn).Commit(req.Context())
+
+	dsTxn, ok := txVal.(datastore.Txn)
+	if !ok {
+		responseJSON(rw, http.StatusBadRequest, errorResponse{ErrInvalidDataStoreTransaction})
```

Comment:
todo: This isn't a `BadRequest` but rather an `InternalServerError` (500). 

Out of scope of this PR but a lot of our returned statuses are wrong like this. I create #3139 to track this. If a panic within the handlers automatically returns a 500 status then it might be better to just do a `must` typecast and let the server panic for cases where a 500 is appropriate.

Reply:
Actually #3139 seems similar / duplicate of what I wanted #3103 to be originally (if i understand correctly) . So I would say that is very much in the scope of this PR. I can change all the required `BadRequests` that need changing if you can guide me to which additional ones should be changed (@nasdf suggested only for the `context` gets+casts originally), I can take care of them in this PR.

Reply:
I agree, allowing a panic or changing to a 500 status would be better.

Reply:
> I agree, allowing a panic or changing to a 500 status would be better.

Just so I understand this, which one are you suggesting:
1) all the `ErrInvalidDataStoreTransaction` to panic?
2) all the `http.StatusBadRequest` to panic?
3) only this line the suggestion is on?

Reply:
Anywhere we do this type cast `txVal.(datastore.Txn)`.

Reply:
Done in commit: `PR(ADHERE): Adhere to code review - mustify datastore.Txn`